packages/launchpad/src/default-wired-launchpad.tsx

```diff
@@ -0,0 +1,79 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { LaunchpadNavigator, useLaunchpad } from '@automattic/data-stores';
+import { useDispatch } from '@wordpress/data';
+import { useEffect } from 'react';
+import Launchpad from './launchpad';
+import { setUpActionsForTasks } from './setup-actions';
+import type { Task } from './types';
+
+type DefaultWiredLaunchpadProps = {
+	siteSlug: string | null;
+	checklistSlug: string;
+	launchpadContext: string;
+};
+
+const DefaultWiredLaunchpad = ( {
+	siteSlug,
+	checklistSlug,
+	launchpadContext,
+}: DefaultWiredLaunchpadProps ) => {
+	const {
+		data: { checklist },
+	} = useLaunchpad( siteSlug, checklistSlug );
+
+	const { setActiveChecklist } = useDispatch( LaunchpadNavigator.store );
+
+	const tasklistCompleted = checklist?.every( ( task: Task ) => task.completed ) || false;
+
+	const tracksData = { recordTracksEvent, checklistSlug, tasklistCompleted, launchpadContext };
+
+	const numberOfSteps = checklist?.length || 0;
+	const completedSteps = ( checklist?.filter( ( task: Task ) => task.completed ) || [] ).length;
+
+	useEffect( () => {
+		// Record task list view as a whole.
+		recordTracksEvent( 'calypso_launchpad_tasklist_viewed', {
+			checklist_slug: checklistSlug,
+			tasks: `,${ checklist?.map( ( task: Task ) => task.id ).join( ',' ) },`,
+			is_completed: tasklistCompleted,
+			number_of_steps: numberOfSteps,
+			number_of_completed_steps: completedSteps,
+			context: launchpadContext,
+		} );
+
+		// Record views for each task.
+		checklist?.map( ( task: Task ) => {
+			recordTracksEvent( 'calypso_launchpad_task_view', {
+				checklist_slug: checklistSlug,
+				task_id: task.id,
+				is_completed: task.completed,
+				context: launchpadContext,
+				order: task.order,
+			} );
+		} );
+	}, [
+		checklist,
+		checklistSlug,
+		completedSteps,
+		numberOfSteps,
+		tasklistCompleted,
+		launchpadContext,
+	] );
+
+	const taskFilter = ( tasks: Task[] ) => {
+		return setUpActionsForTasks( {
+			tasks,
+			siteSlug,
+			tracksData,
+			extraActions: {
+				setActiveChecklist,
+			},
```

Comment:
Minor note for a follow-up: we should allow other overrides via props. As-is, we wouldn't be able to use this from other contexts like Customer Home where we need to inject or override these actions.

Reply:
And maybe not so minor (but still something for a follow-up): the Share Site task relies on a modal, so it doesn't work from the navigator for now. Maybe the thing this changes is how urgent the follow-up is.